### PR TITLE
fix helpers transforming 

### PR DESCRIPTION
**Pull request by fuzhenn**
This is a patch to fix a bug I found in my building flow.

In .babelrc, I add  **transform-proto-to-assign** to fix a bug with class inheritance in IE9/10, as [described here](https://babeljs.io/docs/usage/caveats/#internet-explorer-classes-10-and-below-).

The plugin changes to ```_defaults(obj, defaults)``` instead of ```__proto__``` (IE9/10 doesn't support) for inheritance.
```js
// .babelrc
{
  "presets": [
    ["es2015", { "loose" : true, "modules": false }]
  ],
  "plugins": [
    "transform-proto-to-assign",
    "external-helpers"
  ],
  "ignore": [
    "dist/*.js"
  ]
}
```

However, **transform-proto-to-assign** isn't applied to the external helpers which still use ```__proto__``` for inheritance which will break IE9/10.

This pull request is to fix the problem mentioned above.

Thanks.

### CODE REVIEW

**Comment by fuzhenn** [comment]
ping @hzoo , could you please take some time to have a look at this PR? 
Thanks very much!

**Comment by fuzhenn** [comment]
Some related discussions from [babel](https://github.com/babel/babel/)
https://github.com/babel/babel/issues/1338
https://github.com/babel/babel/issues/3842

This issue may be solved in upcoming 7.x?

**Comment by Andarist** [comment]
This shouldnt happen anymore with babel@7 and its corresponding rollup-plugin-babel@4. I wouldn't like to touch 3.x line, but if you want to rebase this work on `3.x` branch I might do a release for u. You'd also have to tweak the code according to my minor comments.

File: src/index.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { RUNTIME, BUNDLED, HELPERS } from './constants.js';
 const keywordHelpers = [ 'typeof', 'extends', 'instanceof' ];
 
 export default function babel ( options ) {
+	const originalOptions = options;
```
Comment (Andarist) [inline src/index.js]: actually u shouldnt make a copy here, what's left in `options` (after all those `delete`s etc) is exactly what should get passed to babel - you might look how I've refactored it on master, this should be clearer there

File: src/index.js [hunk 2]
```diff
@@ -47,11 +48,13 @@ export default function babel ( options ) {
...
 				const pattern = new RegExp( `babelHelpers\\.(${keywordHelpers.join('|')})`, 'g' );
 
-				const helpers = buildExternalHelpers( externalHelpersWhitelist, 'var' )
+				let helpers = buildExternalHelpers( externalHelpersWhitelist, 'var' )
 					.replace( pattern, 'var _$1' )
 					.replace( /^babelHelpers\./gm, 'export var ' ) +
 					`\n\nexport { ${keywordHelpers.map( word => `_${word} as ${word}`).join( ', ')} }`;
-
+				// Apply babel transforming on helpers, in case ES5/ES3 transform plugins are set in options.
+				helpers = transform( helpers, originalOptions ).code;
```
Comment (Andarist) [inline src/index.js]: when you ditch copying `options` on the top of the plugin you will have to just pass `options` here instead of `originalOptions`

**Comment by Andarist** [comment]
@fuzhenn Cool, thanks for rebasing and adjusting the PR, would u like to add a test for this?

**Comment by fuzhenn** [comment]
Thanks for your review!
I submited another pr #227  to rebase to 3.x branch. C
losing this one.

**Pull request closed without merging**